### ISSUE TITLE
having issues with log in button

### ISSUE THREAD

**Issue by turdferguson420** [geordanr/xwing#608]
### This bug report filer is for First Edition X-Wing only!

**Note:** If you are experiencing problems with the permalink URL when you have **Maul** or **Bomblet Generator** in your list, this is a known issue.  For more information please see https://www.reddit.com/r/XWingTMG/comments/7c4q6h/yet_another_squad_builder_problems_with_maul/

### Expected Behavior

Log In button is not activating

### Actual Behavior

no action when pressed - is there a problem?

### Language

English

**Comment by geordanr** [comment]
What browser (and version) are you using?  There are changes that break older browsers (which is likely to appear on other sites).

**Comment by turdferguson420** [comment]
Hi.

It's Chrome - not sure where I can find the version?

**Comment by geordanr** [comment]
Browse to`chrome://version`; that should tell you what version you're running.

**Comment by turdferguson420** [comment]
Thanks. This is what I came up with:
Google Chrome 79.0.3945.130 (Official Build) (32-bit) (cohort: Stable)
Revision e22de67c28798d98833a7137c0e22876237fc40a-refs/branch-heads/3945@
{#1047}

**Comment by geordanr** [comment]
Strange, that should be compatible.  What OS?

**Comment by turdferguson420** [comment]
OS Windows 10 OS Version 1803 (Build 17134.1246)

**Comment by geordanr** [comment]
Did the login button ever work for you (and only stopped working recently)?  If it never worked, you may have third party cookies disabled, which you'll need to allow.  Otherwise, I'm not sure what's going on; does it work for other browsers on other machines?

**Comment by turdferguson420** [comment]
Correct, the login button worked for at least a couple of years and only
stopped functioning recently. It's a work PC; I have not knowingly changed
any settings. I just tried to access via Explorer and it behaved the same
way.

**Comment by geordanr** [comment]
In Chrome, while you have the builder open, hit F12 to open Developer Tools and click on the "Console" tab, then try to log in.  What error messages (if any) do you see there?

**Comment by turdferguson420** [comment]
Oh, looks like there's a bunch. A copy & paste follows:

2
xwing.min.js?1532065775:5 Error adding card "Heavy Scyk" Interceptor
(title) from C-ROC Cruiser Expansion Pack
VM5:1 Access to XMLHttpRequest at '
https://yasb-backend.herokuapp.com/methods' from origin '
http://geordanr.github.io' has been blocked by CORS policy: No
'Access-Control-Allow-Origin' header is present on the requested resource.
yasb-backend.herokuapp.com/methods:1 Failed to load resource:
net::ERR_FAILED
VM5:1 Access to XMLHttpRequest at 'https://yasb-backend.herokuapp.com/ping'
from origin 'http://geordanr.github.io' has been blocked by CORS policy: No
'Access-Control-Allow-Origin' header is present on the requested resource.
yasb-backend.herokuapp.com/ping:1 Failed to load resource: net::ERR_FAILED
geordanr.github.io/:1 Access to XMLHttpRequest at '
https://yasb-backend.herokuapp.com/settings' from origin '
http://geordanr.github.io' has been blocked by CORS policy: No
'Access-Control-Allow-Origin' header is present on the requested resource.
yasb-backend.herokuapp.com/settings:1 Failed to load resource:
net::ERR_FAILED
/favicon.ico:1 Failed to load resource: the server responded with a status
of 404 (Not Found)

**Comment by geordanr** [comment]
Now I'm really confused; the response I'm getting from the backend does have the `Access-Control-Allow-*` headers:

![image](https://user-images.githubusercontent.com/334985/75214339-0f0a1b00-5742-11ea-95b0-f3e8ff429d2b.png)

Does this happen for you on any other devices?  Or in Incognito Mode (maybe something old is being cached)?

**Comment by turdferguson420** [comment]
To answer your questions:
-I usually go with Incognito, but I tried bringing up the page in reg.
Chrome with same result.
-Tried the page in Chrome on my BlackBerry just now and same issue. Log
button gets highlighted in white when I press on it, but nothing happens
just like on the PC at work.

**Comment by elistevens** [comment]
I also get an error:

```
Access to XMLHttpRequest at 'https://yasb-backend.herokuapp.com/methods' from origin 'http://geordanr.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
yasb-backend.herokuapp.com/methods:1 Failed to load resource: net::ERR_FAILED
geordanr.github.io/:1 Access to XMLHttpRequest at 'https://yasb-backend.herokuapp.com/ping' from origin 'http://geordanr.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
yasb-backend.herokuapp.com/ping:1 Failed to load resource: net::ERR_FAILED
geordanr.github.io/:1 Access to XMLHttpRequest at 'https://yasb-backend.herokuapp.com/settings' from origin 'http://geordanr.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
yasb-backend.herokuapp.com/settings:1 Failed to load resource: net::ERR_FAILED
```

**Comment by geordanr** [comment]
Okay, this is weird.  The error happens for me in incognito mode, but not in my normal browser.

If I try to simulate this with curl by sending different origins, I (correctly) get the `access-control-*` headers.  Simulating access from YASB 1.0:

```
$ curl -I -H 'Origin: https://geordanr.github.io' https://yasb-backend.herokuapp.com/methods
HTTP/1.1 200 OK
Date: Wed, 26 Feb 2020 06:04:37 GMT
Connection: keep-alive
Access-Control-Allow-Origin: https://geordanr.github.io
Access-Control-Allow-Methods: GET, POST, PUT, DELETE
Access-Control-Max-Age: 7200
Access-Control-Allow-Credentials: true
Content-Type: application/json
Content-Length: 50
Vary: Origin
X-Content-Type-Options: nosniff
Server: thin
Via: 1.1 vegur
```

Simulating from YASB 2.0:

```
$ curl -I -H 'Origin: https://raithos.github.io' https://yasb-backend.herokuapp.com/methods
HTTP/1.1 200 OK
Date: Wed, 26 Feb 2020 06:06:14 GMT
Connection: keep-alive
Access-Control-Allow-Origin: https://raithos.github.io
Access-Control-Allow-Methods: GET, POST, PUT, DELETE
Access-Control-Max-Age: 7200
Access-Control-Allow-Credentials: true
Content-Type: application/json
Content-Length: 50
Vary: Origin
X-Content-Type-Options: nosniff
Server: thin
Via: 1.1 vegur
```

Simulating from an unauthorized origin (note lack of headers):

```
$ curl -I -H 'Origin: https://somethingelse.github.io' https://yasb-backend.herokuapp.com/methods
HTTP/1.1 200 OK
Date: Wed, 26 Feb 2020 06:06:23 GMT
Connection: keep-alive
Content-Type: application/json
Content-Length: 50
Vary: Origin
X-Content-Type-Options: nosniff
Server: thin
Via: 1.1 vegur
```

The request headers I'm sending are identical between normal and incognito mode, except for the do not track `dnt` header in normal, which doesn't appear to have any effect when I test with curl (normal is on the left, incognito on the right):

![image](https://user-images.githubusercontent.com/334985/75317159-1d2c6a00-581c-11ea-93bf-724bc1b24c9b.png)

It doesn't matter whether I have cookies set in the `yasb-backend.herokuapp.com` domain, same behavior.

I'm _completely_ stumped.  Anyone else have ideas?

**Comment by geordanr** [comment]
Misclicked close

**Comment by turdferguson420** [comment]
So it's not solvable then? Is there anything I can do on my end?

**Comment by turdferguson420** [comment]
So I'm still trying to get logged in periodically. Chrome is now showing a
button in the browser bar to allow cookies. Got excited when i clicked it
as the log in button highlighted, but it still doesn't take any action when
clicking on it.

Thanks,

Jude
